scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/PipelineSubscriber.kt

```diff
@@ -157,137 +278,30 @@ class PipelineSubscriber(
-            logger.info(
-                "Creating pipeline {pipelineName} version: {pipelineVersion} id: {pipelineId}",
-                metadata.name,
-                metadata.version,
-                metadata.id,
-            )
-        }
-
-        val (pipeline, err) =
-            Pipeline.forSteps(
-                metadata,
-                steps,
-                kafkaProperties,
-                kafkaDomainParams,
-                kafkaConsumerGroupIdPrefix,
-                namespace,
-            )
-        if (err != null) {
-            err.log(logger, Level.ERROR)
-            client.pipelineUpdateEvent(
-                makePipelineUpdateEvent(
-                    metadata = metadata,
-                    operation = PipelineOperation.Create,
-                    success = false,
-                    reason = err.getDescription() ?: "failed to initialize dataflow engine",
-                    timestamp = timestamp,
-                    stream = name,
-                ),
-            )
-            return
         }
 
-        pipeline!! // assert pipeline is not null when err is null
-        if (pipeline.size != steps.size) {
-            pipeline.stop()
-            client.pipelineUpdateEvent(
-                makePipelineUpdateEvent(
-                    metadata = metadata,
-                    operation = PipelineOperation.Create,
-                    success = false,
-                    reason = "failed to create all pipeline steps",
-                    timestamp = timestamp,
-                    stream = name,
-                ),
-            )
-
-            return
-        }
-
-        // This overwrites any previous pipelines with the same id. We can only get here if those previous pipelines
-        // are in a failed state and they are being re-created by the scheduler.
-        pipelines[metadata.id] = pipeline
-        val pipelineStatus: PipelineStatus
-        val errTopics = kafkaAdmin.ensureTopicsExist(steps)
-        if (errTopics == null) {
-            pipelineStatus = pipeline.start()
-        } else {
-            pipelineStatus =
-                PipelineStatus.Error(null)
-                    .withException(errTopics)
-                    .withMessage("kafka streams topic creation error")
-            pipeline.stop()
-        }
-
-        // We don't want to mark the PipelineOperation.Create as successful unless the
-        // pipeline has started. While states such as "StreamStarting" or "StreamStopped" are
-        // not in themselves errors, they are not expected at this stage. If the pipeline
-        // is not running here then it can't be marked as ready.
-        if (pipelineStatus !is PipelineStatus.Started) {
-            pipelineStatus.hasError = true
-        }
-        pipelineStatus.log(logger, Level.DEBUG)
-        client.pipelineUpdateEvent(
-            makePipelineUpdateEvent(
-                metadata = metadata,
+        queues[metadata.id]?.queue?.send(
```

Comment:
Is it worth having this under the mutex? I'm not sure what happens if the queue disappears between fetching `queues[metadata.id]` and calling `.queue.send` on it.